### PR TITLE
fix: replace tree with code mirror for metadata view

### PR DESCRIPTION
**Pull request by keita-determined**
## Description

Replace metadata tree view with code mirror to show metadata. 

### Reason

The tree view does not handle all edge cases to show metadata such as nested array, objects in array etc. It would be difficult to handle all edge cases and maintain it, so proposed to use [Code Mirror Editor](https://uiwjs.github.io/react-codemirror/) to show metadata ([Slack thread](https://hpe-aiatscale.slack.com/archives/C06G9QEKUJZ/p1724269555897989)). Especially we already use it in `Code` tab in the experiment/run detail page. Code mirror editor has features to expand/collapse code block and search metadata with regex. Also, it's able to show any complex or big metadata easily, and it's very easy to maintain it because of out of box features. That's why we chose code mirror editor.
Other option was to use JSON viewer, but that means we need to add another dependency which we didn't prefer.

#### Bug in the tree view (Before this PR)

- not correctly show metadata
- every time i click a expand button, metadata is getting bigger and bigger (see the scroll bar in the video)

https://github.com/user-attachments/assets/12ba8b9a-0145-4d95-971d-022912dd54f9

### After

https://github.com/user-attachments/assets/abec8323-0eaf-4f36-991e-5fcb8520c199

## Test Plan

- add complex metadata such as nested array or objects in array etc for a run
- go to the run detail page
- Check if metadata is visible at the bottom
- Check if code blocks are expandable/collapsable
- Check if theres no regressions in Code tab in experiment/run detail page as well
  - code highlight

## Checklist

- [x] Changes have been manually QA'd
- [ ] New features have been approved by the corresponding PM
- [ ] User-facing API changes have the "User-facing API Change" label
- [ ] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses have been included for new code which was copied and/or modified from any external code

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webui/react/package.json [hunk 1]
```diff
@@ -32,11 +32,7 @@
     "unimported": "unimported"
   },
   "dependencies": {
-    "@codemirror/lang-markdown": "^6.2.5",
-    "@codemirror/lang-python": "^6.1.6",
-    "@codemirror/legacy-modes": "^6.3.3",
     "@glideapps/glide-data-grid": "^6.0.3",
-    "@uiw/react-codemirror": "^4.22.00",
```
Comment (keita-determined) [inline webui/react/package.json:39]: we moved unnecessary dependencies. code mirror deps are installed through hew library

File: webui/react/src/components/Metadata.module.scss [hunk 2]
```diff
@@ -1,5 +1,3 @@
 .base {
-  :global(.ant-tree-node-content-wrapper-close) {
-    font-weight: bold;
-  }
+  margin-bottom: 32px;
```
Comment (EmilyBonar) [inline webui/react/src/components/Metadata.module.scss:2]: nit: should use the spacing variables. 32px is also pretty large, i suggest `var(--spacing-xl-2)` which is 20px

**Review by EmilyBonar — APPROVED** [review]
LGTM, just a note on spacing

File: webui/react/src/components/Metadata.module.scss (hunk 2 above)
Comment (keita-determined) [reply webui/react/src/components/Metadata.module.scss]: fixed

**Pull request merged**